packages/vite/src/node/plugins/modulePreloadPolyfill.ts

```diff
@@ -13,11 +14,11 @@ export function modulePreloadPolyfillPlugin(config: ResolvedConfig): Plugin {
     name: 'vite:modulepreload-polyfill',
     resolveId(id) {
       if (id === modulePreloadPolyfillId) {
-        return id
+        return resolvedModulePreloadPolyfillId
```

Comment:
This is to avoid breaking change for this usage.
https://vitejs.dev/guide/backend-integration.html#backend-integration:~:text=If%20you%20haven%27t%20disabled%20the%20module%20preload%20polyfill%2C%20you%20also%20need%20to%20import%20the%20polyfill%20in%20your%20entry